client/src/elm/Backend/Update.elm

```diff
@@ -4271,15 +4280,49 @@ generatePrenatalAssessmentMsgs currentDate language isChw activePage updateAsses
                                     reportedDiagnoses =
                                         List.filter (\diagnosis -> List.member diagnosis targetDiagnoses) addedDiagnoses
                                             |> EverySet.fromList
+
+                                    diabetesDiagnoses =
+                                        EverySet.toList reportedDiagnoses
+                                            |> List.any
+                                                (\diagnosis ->
+                                                    List.member diagnosis Pages.Prenatal.Utils.diabetesDiagnoses
+                                                )
+
+                                    rhNegativeDiagnosis =
+                                        EverySet.member DiagnosisRhesusNegative reportedDiagnoses
                                 in
-                                if not <| EverySet.isEmpty reportedDiagnoses then
-                                    [ Backend.PrenatalEncounter.Model.SetPastPrenatalDiagnoses reportedDiagnoses
-                                        |> Backend.Model.MsgPrenatalEncounter originatingEncounterId
-                                        |> App.Model.MsgIndexedDb
-                                    ]
+                                if EverySet.isEmpty reportedDiagnoses then
+                                    []
 
-                                else
+                                else if
+                                    (-- Reporting back about previous diagnois results in hospital referral
+                                     -- at Next steps. For Diabetes, we have logic saying that when it's first
+                                     -- diagnosed, patient is referred to hospital.
+                                     -- On next occasions, no next steps are required.
+                                     -- Therefore, if we know that Diabetes was already diagnosed, we will not
+                                     -- report back about this diagnosis, to prevent unnecessary referral to hospital.
```

Comment:
```suggestion
                                    (-- Reporting back about previous diagnosis results in hospital referral
                                     -- at Next steps. For Diabetes, we have logic saying that when it's first
                                     -- diagnosed, the patient is referred to hospital.
                                     -- On next occasions, no next steps are required.
                                     -- Therefore, if we know that Diabetes was already diagnosed, we will not
                                     -- report back about this diagnosis, to prevent unnecessary referral to the hospital.
```